pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -39,14 +41,52 @@ func (strategy) AllowUnconditionalUpdate() bool {
 
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
 func (strategy) PrepareForCreate(obj runtime.Object) {
-	_ = obj.(*api.DeploymentConfig)
-	// TODO: need to ensure status.latestVersion is not set out of order
+	dc := obj.(*api.DeploymentConfig)
+	dc.Generation = 1
+	dc.Status = api.DeploymentConfigStatus{}
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (strategy) PrepareForUpdate(obj, old runtime.Object) {
-	_ = obj.(*api.DeploymentConfig)
-	// TODO: need to ensure status.latestVersion is not set out of order
+	newDc := obj.(*api.DeploymentConfig)
+	oldDc := old.(*api.DeploymentConfig)
+
+	oldVersion := oldDc.Status.LatestVersion
+	newVersion := newDc.Status.LatestVersion
+	newStatus := newDc.Status
+
+	// Check for imagechange controller change
+	isImageControllerChange := deployutil.IsImageChangeControllerChange(*newDc, *oldDc)
+
+	// Persist status
+	newDc.Status = oldDc.Status
+
+	// If this update comes from the imagechange controller, tolerate status detail updates.
+	if isImageControllerChange {
+		newDc.Status.Details = newStatus.Details
```

Comment:
This is a sticky bit. Yes, it could be faked and impersonating the image change controller could potentially block new deployments from config triggers (https://github.com/openshift/origin/blob/cc51f3b1a60c943449795ec2cefc926696590d46/pkg/deploy/controller/configchange/controller.go#L100-L106) but that's a racy corner case. Eventually, we will rework the way the image change and config change controllers work by trashing the generator they both use which will vanish this concern.